File: src/main/java/com/horus/formataArquivoHorus/model/entity/CompetenceHorus.java
package com.horus.formataArquivoHorus.model.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.util.List;

@jakarta.persistence.Entity
@Table(name = "competence_horus")
@Data
@EqualsAndHashCode
public class CompetenceHorus {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "competence")
    private String competence;

    @OneToMany(mappedBy = "competenceHorus")
    private List<Product> products;

    @ManyToOne
    @JoinColumn(name = "sector_fk")
    @JsonIgnore
    private Sector sector;

}

File: view/view-farma/src/app/interface/competence-horus.ts
import { Product } from "./product"; // Importe a interface de Product
import { Sector } from './sector'; // Importe a interface de Sector

export interface CompetenceHorus {
  id?: number; // Em TypeScript, 'number' corresponde a Long e outras formas de números inteiros
  competence?: string;
  products?: Product[]; // Relacionamento OneToMany com Product
  sector?: Sector; // Relacionamento ManyToOne com Sector
}